### DISCUSSION TITLE
Command wheel for pac_events

### DISCUSSION THREAD

**Discussion by TW1STaL1CKY** [CapsAdmin/pac3#990] [Ideas]
Creating an issue so the idea isn't lost in the Discord

A concommand (+pac_wheel/-pac_wheel ?) that displays a command wheel HUD showing you "command" events you could trigger, in the same way you can do stuff using the Expressions wheel from VRChat.

The command wheel would populate itself by scanning through the localplayer's PAC for "command" events that meet criteria.
Each command option needs a way to either fire one-off (runs "pac_event thing") or toggle (runs "pac_event thing 2"). Maybe via single-click and double-click?

Two new settings can be added to command events for managing the command wheel:
- **show on command wheel** - true/false - enables the command to appear on the wheel
- **command wheel folder** - string - places commands inside a folder within the wheel for organisation

Hook could be added for developers:
- **PreShowCommandWheel** - runs to check if the wheel should render (eg. server owner might want to disable the use of the wheel, or someone might want to make their own version of the wheel)

**Comment by thegrb93** [comment]
I have used this for my starfall emote chip. Adding to pac would be a great addition.

**Comment by CapsAdmin** [comment]
> I have used this for my starfall emote chip. Adding to pac would be a great addition.

maybe you can add it to pac?

**Comment by thegrb93** [comment]
> > I have used this for my starfall emote chip. Adding to pac would be a great addition.
> 
> maybe you can add it to pac?

I can add the graphics and interface if you can add the new event type and table of them that the selector can read.

**Reply by PapersTheKingOfPapers** [reply]
I have no idea how to code lua, but the wheel buttons should basically just execute the command attached to it.

**Reply by thegrb93** [reply]
It's already been added

**Reply by PapersTheKingOfPapers** [reply]
> It's already been added

Where?

**Comment by CapsAdmin** [comment]
A minimal viable version was added for testing, so it's open for feedback (well everything is always open for feedback, but this is on the radar at the moment)

I think my own feedback after using a bit it is that we probably need to add a "hide in event wheel" property to events. Some people make events that are meant to be triggered by other events, so that they are kind of "internal" features of the outfit and not public.

**Reply by thegrb93** [reply]
Probably also need some user instruction for setting up the event selector.

**Reply by TW1STaL1CKY** [reply]
I've given some ideas for settings on event parts and a hook in the original post 😃
